web/geo/generateZoneBoundingBoxes.ts

```diff
@@ -1,36 +1,50 @@
+import fs from 'node:fs';
 import path from 'node:path';
 import { fileURLToPath } from 'node:url';
-import { mergeZones } from '../scripts/generateZonesConfig.js';
+import yaml from 'js-yaml';
 import { saveZoneYaml } from './files.js';
 import { getJSON } from './utilities.js';
-import { WorldFeatureCollection } from './types.js';
+import { WorldFeatureCollection, ZoneConfig } from './types.js';
 import { Position } from '@turf/turf';
 
 const inputArguments = process.argv.slice(2);
 
 const zonesGeo: WorldFeatureCollection = getJSON(
   path.resolve(fileURLToPath(new URL('world.geojson', import.meta.url)))
 );
-const zones = mergeZones();
 
 if (inputArguments.length <= 0) {
   console.error(
-    'ERROR: Please add a zoneName parameter ("ts-node generateZoneBoundingBoxes.ts DE")'
+    'ERROR: Please add a zoneName parameter ("ts-node --esm generateZoneBoundingBoxes.ts DE")'
   );
   process.exit(1);
 }
 
 const zoneKey = inputArguments[0];
 
-if (!(zoneKey in zones)) {
+const zonePath = path.resolve(
+  fileURLToPath(new URL(`../../config/zones/${zoneKey}.yaml`, import.meta.url))
+);
+const zoneConfig = yaml.load(fs.readFileSync(zonePath, 'utf8')) as ZoneConfig;
+
+if (!zoneConfig) {
   console.error(`ERROR: Zone ${zoneKey} does not exist in configuration`);
   process.exit(1);
 }
 
-zonesGeo.features = zonesGeo.features.filter((d) => d.properties.zoneName === zoneKey);
+let zoneFeatures = zonesGeo.features.filter((d) => d.properties.zoneName === zoneKey);
+let isAggregate = false;
+if (zoneFeatures.length <= 0) {
+  console.info(`Zone ${zoneKey} does not exist in geojson, using subzones instead`);
+  isAggregate = true;
+  zoneFeatures = zonesGeo.features.filter((d) => {
+    return d.properties.countryKey === zoneKey;
+  });
+}
+zonesGeo.features = zoneFeatures;
 
 let allCoords = [];
```

Comment:
Can't add a full suggestion as you didn't change the lines below but I think we should do something like this:
```ts
let allCoords: Position[] = [];
let boundingBoxes: { [key: string]: number[][] } = {};

for (const zone of zonesGeo.features) {
  allCoords = [];
  const geometryType = zone.geometry.type;
  for (const coords1 of zone.geometry.coordinates) {
    for (const coord of coords1[0]) {
      allCoords.push(coord as Position);
    }
  }
```
That way we get rid of a type error.

Reply:
Nice, I was a bit annoyed by the type error, so thanks for the suggestion! I had to do some additional changes on top of your suggestion here to avoid additional errors :)